src/Xamarin.Android.Build.Tasks/Utilities/ManagedResourceParser.cs

```diff
@@ -214,13 +216,13 @@ public CodeTypeDeclaration Parse (string resourceDirectory, IEnumerable<string>
 						name = name.Replace ("android:", string.Empty);
 						var element = publicXml?.XPathSelectElement ($"/resources/public[@name='{name}']") ?? null;
 						value = Convert.ToInt32 (element?.Attribute ("id")?.Value ?? "0x0", fromBase: 16);
-						
 					} else {
 						CodePrimitiveExpression initExpression = fields [i].InitExpression as CodePrimitiveExpression;
 						value = Convert.ToInt32 (initExpression.Value);
 					}
-					arrayValues.Add (value, fields [i]);
+					arrayValues.Add ((Key: value, Value: fields [i]));
 				}
+				arrayValues.Sort (new CompareTuple ());
```

Comment:
This is really minor, but it seems like the `CompareTuple` could be stored in a static field and reused. Seems like this loop could get called a lot?